### PR TITLE
Changes to allow login/password resets with email

### PR DESCRIPTION
**Pull request by doubt72**
What it says.

ChangeLog-Entry: Allow sign in with email address in oc-id

### CODE REVIEW

File: src/oc-id/app/views/password_resets/new.html.erb [hunk 1]
```diff
@@ -3,11 +3,11 @@
     <h1>Forgot Your Password?</h1>
   </div>
   <div>
-    <p>Enter your hosted Chef username and we'll send you a link to reset it.</p>
+    <p>Enter your hosted Chef username or email address and we'll send you a link to reset it.</p>
```
Comment (smith) [inline src/oc-id/app/views/password_resets/new.html.erb]: This should not say "hosted", let's take it out.

**Comment by smith** [comment]
We should probably say "username, email address, or password" here: https://github.com/chef/chef-server/blob/38beb145694953fac341d2944da2786a8e39deb2/src/oc-id/app/controllers/sessions_controller.rb#L14

**Comment by smith** [comment]
These also should mention email: https://github.com/chef/chef-server/blob/38beb145694953fac341d2944da2786a8e39deb2/src/oc-id/config/locales/en.yml#L30-L34

The previous one I mentioned should be an i18n tag as well, rather than hard-coded into the controller.

**Comment by smith** [comment]
Password reset works for me, but sign in does not. The sign in form POSTs to `auth_chef_callback_path`. Does this need to have chef/omniauth-chef#11pulled in to work?

**Comment by doubt72** [comment]
Yeah, it won't work until omniauth is updated; reset uses an internal path, but login is via omniauth.  I've been testing it by changing the gemfile locally to use path instead of github, or you could change the branch now that it's up.

**Comment by smith** [comment]
Ok I'll try that.

#553 removes references to "hosted" and puts them into i18n, so you could rebase against that once it's ok'd and merged.

**Comment by smith** [comment]
Tried this running with chef/omniauth-chef#11 and it worked so :+1: once we get that released and updated here.

**Comment by smith** [comment]
Updated to latest chef/omniauth-chef and rebased. @raskchanky @doubt72 would you mind taking a final look?

**Comment by raskchanky** [comment]
:+1:

**Comment by doubt72** [comment]
:+1: ...assuming the checks are related to omniauth gem not being updated yet.

**Pull request merged**